server/render/index.js

```diff
@@ -226,3 +199,57 @@ export function serverRenderError( err, req, res, next ) {
 
 	next();
 }
+
+/**
+ * A generic middleware that identifies if the current request fits the conditions for being SSRable.
+ *
+ * Warning: Having serverSideRender=true is not sufficient. The app-level logic needs be applied before truly SSRing.
```

Comment:
That reads a bit confusing to me, since it's not entirely clear whose responsibility it is to set what, and where. Furthermore, it might actually invite people to add section-specific logic here (which we want to avoid). (Also see my other comment, https://github.com/Automattic/wp-calypso/pull/25554/files#r198841859)

I think the JSDoc should make clear:
* This is a fallback middleware that sets `context.serverSideRender` early on
* If consumers need to override this at section level, they should set any section specific criteria by adding section specific middleware
* Even then, there are app level criteria that take precedence over `context.serverSideRender === true`